lib/travis/build/addons/sauce_connect.rb

```diff
@@ -15,6 +15,15 @@ def before_before_script
           sh.export 'SAUCE_USERNAME', username, echo: false if username
           sh.export 'SAUCE_ACCESS_KEY', access_key, echo: false if access_key
 
+          direct_domains_arg = direct_domains ? "-D #{direct_domains}" : ''
+          sh.export 'SAUCE_DIRECT_DOMAINS', direct_domains_arg, echo: false if direct_domains_arg
```

Comment:
Note that this line will export `SAUCE_DIRECT_DOMAINS` as an empty string, since `''` is a truth-y value. (This is not a problem here, I'm just pointing it out.)

Perhaps we can roll these 2 statements into one `if` statement to make the logic clearer:

```ruby
if direct_domains
  sh.export 'SAUCE_DIRECT_DOMAINS', "-D #{direct_domains}", echo: false
end
```